### PR TITLE
Stop using DiffHelpers

### PR DESCRIPTION
**Pull request by pirj**
Depends on:
 - [x] https://github.com/rspec/rspec-core/pull/2833
 - [x] https://github.com/rspec/rspec-expectations/pull/1268
 - [x] https://github.com/rspec/rspec-mocks/pull/1384
 - [ ] https://github.com/rspec/rspec-dev/pull/278

### CODE REVIEW

File: spec/rspec/support/differ_spec.rb [hunk 1]
```diff
@@ -1,554 +0,0 @@
-require 'ostruct'
-require 'timeout'
-require 'rspec/support/spec/string_matcher'
-
-module RSpec
```
Comment (JonRowe) [inline spec/rspec/support/differ_spec.rb:5]: Please restore this spec, and just remove the use of `diff_helpers`.

**Review by JonRowe — CHANGES_REQUESTED** [review]
The differ spec is there to provide specs for the differ, it needs to remain.

**Comment by pirj** [comment]
```
Coverage report generated for RSpec to /home/runner/work/rspec-support/rspec-expectations/coverage. 3116 / 3596 LOC (86.65%) covered.
```
I don't get it. Locally `rspec-expectations`'s `4-0-dev` branch:
```
 46 files in total.
2793 relevant lines, 2763 lines covered and 30 lines missed. ( 98.93% ) 
```

**Comment by pirj** [comment]
`Coverage report generated for RSpec to /home/runner/work/rspec-expectations/rspec-expectations/coverage. 2905 / 3059 LOC (94.97%) covered.` according to https://github.com/rspec/rspec-expectations/runs/1626068228?check_suite_focus=true

**Review by JonRowe — APPROVED** [review]

**Pull request merged**